igor-core/src/main/java/com/netflix/spinnaker/igor/build/model/GenericArtifact.java

```diff
@@ -41,6 +41,13 @@ public GenericArtifact(String type, String name, String version, String referenc
     this.name = name;
     this.version = version;
     this.reference = reference;
+    this.fileName = String.format("%s:%s", name, version);
+  }
+
+  public GenericArtifact withUrl(String url) {
+    this.displayPath = url;
+    this.url = url;
+    return this;
```

Comment:
Replace with Lombok's `@Wither` annotation.

Reply:
In general, I agree with the suggestion, just note that @nimakaviani is actually setting two different fields within this method.

Reply:
In general, I agree with the suggestion, just note that @nimakaviani is actually setting two different fields within this method (which maybe an anti-pattern, I guess).

Reply:
yeah it is setting two fields. will keep it as is.

Reply:
I think that is a bad idea, as you're breaking the wither-pattern. If you really want to set two fields in the same with-method, then call it something else.

Reply:
yeah fair point. on a second look, it made set them separately.